bertopic/plotting/_datamap.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+import pandas as pd
+from typing import List, Union
+from umap import UMAP
+from warnings import warn
+try:
+    import datamapplot
+    from matplotlib.figure import Figure
+except ImportError:
+    warn("Data map plotting is unavailble unless datamapplot is installed.")
+
+def visualize_document_datamap(topic_model,
+                               docs: List[str],
+                               embeddings: np.ndarray = None,
+                               reduced_embeddings: np.ndarray = None,
+                               custom_labels: Union[bool, str] = False,
+                               title: str = "Documents and Topics",
+                               sub_title: Union[str, None] = None,
+                               width: int = 1200,
+                               height: int = 1200,
+                               **datamap_kwds) -> Figure:
```

Comment:
This throws an error if Figure was not imported from matpotlib if it was not installed. Which is why the testing pipeline failed.

Reply:
That makes sense; my tests ran locally, but of course matplotlib was installed. I think I can wrap the import in a try accept and have a dummy class for other case.

Reply:
This should work now. See [f8104c2](https://github.com/MaartenGr/BERTopic/pull/1750/commits/f8104c202ef57cddfac6f1da287521f316ee9a59).